CDCgov/prime-reportstream
# Engagement/jessica/15406 cli download report

This PR adds a CLI command to download a report from a specified environment while optionally (unless it is prod) removing PII.

Test Steps:
1. Get a report id to query for, make sure it is returned with and without the PII removed as you specify in the command
2. Get a report id that is for an HL7 report and make sure it returns a message saying it is the wrong type. 
3. Try to download a prod report with remove-pii set to false and make sure it does not allow this.
Example query: downloadMessage -o /Users/jessicawillis/git/newnew/prime-reportstream/prime-router/src/testIntegration/resources/datatests/FHIR_to_HL7/downloaded.fhir --env local -r 777fc1d7-dd51-4f01-ad56-6e51b4721caa --remove-pii false

## Changes
- CLI command added to download a report

## Checklist

### Testing

### Process

## Linked Issues
- Fixes #issue

## To Be Done
*Create GitHub issues to track the work remaining, if any*
- #issue

## Specific Security-related subjects a reviewer should pay specific attention to
- Does this PR introduce new endpoints?
    - new endpoint A
    - new endpoint B
- Does this PR include changes in authentication and/or authorization of existing endpoints?
- Does this change introduce new dependencies that need vetting?
- Does this change require changes to our infrastructure?
- Does logging contain sensitive data?
- Does this PR include or remove any sensitive information itself?

If you answered '_yes_' to any of the questions above, conduct a detailed Review that addresses at least:

- What are the potential security threats and mitigations? Please list the _STRIDE_ threats and how they are mitigated
    - **S**poofing (faking authenticity)
        - Threat _T_, which could be achieved by _A_, is mitigated by _M_
    - **T**ampering (influence or sabotage the integrity of information, data, or system)
    - **R**epudiation (the ability to dispute the origin or originator of an action)
    - **I**nformation disclosure (data made available to entities who should not have it)
    - **D**enial of service (make a resource unavailable)
    - **E**levation of Privilege (reduce restrictions that apply or gain privileges one should not have)
- Have you ensured logging does not contain sensitive data?
- Have you received any additional approvals needed for this change?
